### PR TITLE
Fix Pressing enter in console with console run keystroke set to enter creates a newline and runs

### PR DESCRIPTION
**Pull request by FoSuCloud**
## References

Fixes: https://github.com/jupyterlab/jupyterlab/issues/15835

## Code changes

1. Add extensions options in the getEditorOptions function and provide them to the InputArea of ​​the cells module.
2. Only when INTERACTION_MODE is terminal, prevent the default event of the enter key.

## User-facing changes

before change

https://github.com/jupyterlab/jupyterlab/assets/49218295/0e78f27d-8b56-4bbe-b295-ac6258751249

after change

https://github.com/jupyterlab/jupyterlab/assets/49218295/296e2533-ee3a-49a6-801b-c49476ed41d3

## Backwards-incompatible changes

none

### CODE REVIEW

**Comment by FoSuCloud** [comment]
@meeseeksdev tag bug

File: packages/cells/src/widget.ts [hunk 1]
```diff
@@ -308,6 +309,13 @@ export class Cell<T extends ICellModel = ICellModel> extends Widget {
     return this._editorConfig;
   }
 
+  /**
+   * Editor Extensions
+   */
+  get editorExtensions(): Extension[] {
```
[the remark this answers is not in the document]
Comment (FoSuCloud) [reply packages/cells/src/widget.ts:315]: You are right, I will modify it

**Comment by krassowski** [comment]
Re-running tests to see if the `Completer › Console › Open completer on console` and `Completer › Console › Filter console completer suggestions by typing` might be related.

**Comment by FoSuCloud** [comment]
I don’t have this problem in my local macos environment. Can I re-execute the test and see the results?
截屏2024-02-26 21 57 10

**Comment by FoSuCloud** [comment]
> Re-running tests to see if the `Completer › Console › Open completer on console` and `Completer › Console › Filter console completer suggestions by typing` might be related.

@krassowski  please re-execute the test

**Comment by krassowski** [comment]
I did and it still fails. I re-run it again now and if it still fails it would be a strong argument that we need to change something more here.

**Comment by krassowski** [comment]
Yup, still failing, the exact two jobs:
``
    [jupyterlab] › test/jupyterlab/completer.test.ts:170:9 › Completer › Console › Open completer on console 
    [jupyterlab] › test/jupyterlab/completer.test.ts:178:9 › Completer › Console › Filter console completer suggestions by typing 
```

**Comment by FoSuCloud** [comment]
It does look like there is a problem, I will check these two test cases

**Comment by FoSuCloud** [comment]
It looks like I made a mistake and didn't limit preventDefault.
The correct approach is to only block the default event when selecting "Execute with Enter"

**Comment by FoSuCloud** [comment]
@krassowski There is only one test question left "test/documentation/overview.test.ts:24:7 › Overview › Open tabs", 
but it seems to have nothing to do with this PR

**Review by krassowski — APPROVED** [review]
Thank you @FoSuCloud!

> For future, it is easier to review PRs which include a test case. In this instance I would go for galata (playwright) integration test.

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.1.x

**Pull request merged**